pkg/api/graph/types.go

```diff
@@ -47,6 +51,72 @@ const (
 	ReferencedImageLayerGraphEdgeKind
 )
 
+var kindToGraphKind = map[reflect.Type]interface{}{}
+
+func init() {
+	kindToGraphKind[reflect.TypeOf(&image.ImageStream{})] = ImageStreamGraphKind
+	kindToGraphKind[reflect.TypeOf(&image.Image{})] = ImageGraphKind
+	kindToGraphKind[reflect.TypeOf(&build.BuildConfig{})] = BuildConfigGraphKind
+	kindToGraphKind[reflect.TypeOf(&build.Build{})] = BuildGraphKind
+	kindToGraphKind[reflect.TypeOf(&deploy.DeploymentConfig{})] = DeploymentConfigGraphKind
+	kindToGraphKind[reflect.TypeOf(&kapi.Service{})] = ServiceGraphKind
+	kindToGraphKind[reflect.TypeOf(&kapi.Pod{})] = PodGraphKind
+	kindToGraphKind[reflect.TypeOf(&kapi.ReplicationController{})] = ReplicationControllerGraphKind
+}
+
+func GetUniqueNamespaceNodeName(obj runtime.Object) UniqueName {
+	return getUniqueNodeName(obj, true)
+}
+
+func GetUniqueRootScopedNodeName(obj runtime.Object) UniqueName {
+	return getUniqueNodeName(obj, false)
+}
+
+func getUniqueNodeName(obj runtime.Object, namespaced bool) UniqueName {
+	objType := reflect.TypeOf(obj)
+	graphKind, exists := kindToGraphKind[objType]
+
+	if !exists {
+		panic(fmt.Sprintf("no graphKind registered for %v", objType))
+	}
+
+	meta, err := kapi.ObjectMetaFor(obj)
+	if err != nil {
+		panic(err)
+	}
+
+	if namespaced {
+		return UniqueName(fmt.Sprintf("%d|%s/%s", graphKind, meta.Namespace, meta.Name))
+	}
+
+	return UniqueName(fmt.Sprintf("%d|%s", graphKind, meta.Name))
+}
+
+func ImageStreamNodeName(o *image.ImageStream) UniqueName {
+	return GetUniqueNamespaceNodeName(o)
+}
+
+type ImageStreamNode struct {
+	Node
+	*image.ImageStream
+}
+
+func (n ImageStreamNode) Object() interface{} {
+	return n.ImageStream
+}
+
+func (n ImageStreamNode) String() string {
+	return fmt.Sprintf("<ImageStream %s/%s>", n.Namespace, n.Name)
```

Comment:
Do you want to be consistent with the other funcs as far as capitalization goes? ImageStream -> imagestream? Personally I think CamelCase looks better (easier to distinguish separate terms).